### PR TITLE
feat: add asinh function to math.

### PR DESCRIPTION
**Pull request by CusiniM**
Adding hyperbolic sine because it's needed in https://github.com/GEOS-DEV/GEOS/pull/3280

### CODE REVIEW

File: src/math.hpp [hunk 1]
```diff
@@ -1065,6 +1065,36 @@ __half2 log( __half2 const x )
...
+double asinh( T const x )
+{
+#if defined(LVARRAY_DEVICE_COMPILE)
+  return ::asinh( double( x ) );
+#else
+  return std::asinh( x );
+#endif
+}
+
+#endif
```
Comment (CusiniM) [inline src/math.hpp]:
```suggestion
```

**Comment by CusiniM** [comment]
> Adding hyperbolic sine because it's needed in [GEOS-DEV/GEOS#3280](https://github.com/GEOS-DEV/GEOS/pull/3280)

I have not added half precision versions of these because I would have to implement them. I guess that they could be added if really necessary.

File: unitTests/testMath.cpp [hunk 2]
```diff
@@ -199,6 +211,27 @@ TYPED_TEST( TestMath, exponential )
   this->exponential();
 }
 
+using TestComplexMathTypes = ::testing::Types<
```
Comment (corbett5) [inline unitTests/testMath.cpp:221]: Is this just to avoid using `__half`?

**Review by corbett5 — APPROVED** [review]

**Comment by corbett5** [comment]
> > Adding hyperbolic sine because it's needed in [GEOS-DEV/GEOS#3280](https://github.com/GEOS-DEV/GEOS/pull/3280)
> 
> I have not added half precision versions of these because I would have to implement them. I guess that they could be added if really necessary.

Yeah I only added half precision support to testing out on the Chevron finite difference code, I don't think we have to maintain it for everything.

File: unitTests/testMath.cpp (hunk 2 above)
Comment (CusiniM) [reply unitTests/testMath.cpp:221]: yes.

**Pull request merged**